Add column publication status for cloud

garaimanoj opened the pull request:
Resolved https://stfc.atlassian.net/browse/GT-1222

http://host-172-16-103-240.nubes.stfc.ac.uk/publishing/cloud/
<img width="955" height="656" alt="image" src="https://github.com/user-attachments/assets/c9016f9f-2571-4be3-8720-04a5824244e1" />

garaimanoj committed:
Add column publication status for cloud
Added days parameter for flexibility.
Removed redundant elif. Handling future date at the beginning.
Used diff_days instead of timedelta comparison for clarity.
Used dict.update() for cleaner dictionary updates.
Used f-strings for better readability.

Refactoring

Add new line

tofu-rocketry reviewed (changes requested):
The commit messages need improvement. I expect to see "why" the changes have been made. For this change, particularly why the `update_dict_stdout_and_returncode` function was overhauled like that.

garaimanoj commented:
> The commit messages need improvement. I expect to see "why" the changes have been made. For this change, particularly why the `update_dict_stdout_and_returncode` function was overhauled like that.

- Added days parameter for flexibility.
- Removed redundant elif. Handling future date at the beginning.
- Used diff_days instead of timedelta comparison for clarity.
- Used dict.update() for cleaner dictionary updates.
- Used f-strings for better readability.

tofu-rocketry commented:
Thanks, that's useful context. Can you squash the commits and add those points to the commit message, please?

As an aside - one little quirk of this new system we should keep in mind, compared to the old pub/sync system, is that the status is based on when the page is accessed, whereas the old system assembled static pages so wouldn't raise a warning if the update cron wasn't running/working. This is probably a good thing as it means we don't give a false impression that publishing is ok, but could mean we end up with lots of sites panicking at once.

tofu-rocketry commented:
@garaimanoj Don't forget to delete your branches once a PR's merged.
